pkg/api/v1beta3/types.go

```diff
@@ -206,7 +206,10 @@ type HostPathVolumeSource struct {
 	Path string `json:"path" description:"path of the directory on the host"`
 }
 
-type EmptyDirVolumeSource struct{}
+type EmptyDirVolumeSource struct {
+	// Optional: make this empty directory be backed by memory.
+	Tmpfs bool `json:"tmpfs" description:"use tmpfs instead of disk"`
```

Comment:
To clarify, I am proposing "" as the default.  I don't think this will
realisitically ever change between API versions (e.g. v2beta1 is not going
to make the default medium be explicity tmpfs), so converting "" (versioned
zero-value) into "Default" internal is pointless, especially because it has
to happen in every versioned API.

On Tue, Mar 10, 2015 at 9:26 PM, Victor Marmol <notifications@github.com>
wrote:

> In pkg/api/v1beta3/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5166#discussion_r26188270>
> :
>
> > @@ -206,7 +206,10 @@ type HostPathVolumeSource struct {
> >  	Path string `json:"path" description:"path of the directory on the host"`
> >  }
> >
> > -type EmptyDirVolumeSource struct{}
> > +type EmptyDirVolumeSource struct {
> > +	// Optional: make this empty directory be backed by memory.
> > +	Tmpfs bool `json:"tmpfs" description:"use tmpfs instead of disk"`
>
> I'm +1 on "default" and "memory". Default will let us make it
> implementation-defined.
>
> We've started some the hardware detection in cAdvisor with CPU, caches,
> and FS.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5166/files#r26188270>
> .
>